fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/simulation/perception/index/objects/VehicleBoundingBox.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2023 Fraunhofer FOKUS and others. All rights reserved.
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contact: mosaic@fokus.fraunhofer.de
+ */
+
+package org.eclipse.mosaic.fed.application.ambassador.simulation.perception.index.objects;
+
+import org.eclipse.mosaic.lib.math.Vector3d;
+import org.eclipse.mosaic.lib.math.VectorUtils;
+import org.eclipse.mosaic.lib.spatial.Edge;
+
+import com.google.common.collect.Lists;
+
+import java.util.List;
+
+/**
+ * An object representing a vehicles' 2D bounding box.
+ * <pre>
+ *   frontRightCorner----------rightEdge----------backRightCorner
+ *                  |                             |
+ *   Heading <== frontEdge                     backEdge
+ *                  |                             |
+ *    frontLeftCorner----------leftEdge-----------backLeftCorner
+ * </pre>
+ */
+public class VehicleBoundingBox implements SpatialObjectBoundingBox {
+    private final List<Vector3d> allCorners;
+    private final Vector3d frontRightCorner;
+    private final Vector3d backRightCorner;
+    private final Vector3d backLeftCorner;
+    private final Vector3d frontLeftCorner;
+
+    private final List<Edge<Vector3d>> allEdges;
+    private final Edge<Vector3d> frontEdge;
+    private final Edge<Vector3d> rightEdge;
+    private final Edge<Vector3d> backEdge;
+    private final Edge<Vector3d> leftEdge;
```

Comment:
for me it would be fine if these fields are `public final` and remove all getters, since they are not helpful in this case.